src/Microsoft.OData.Core/UriParser/Parsers/SelectExpandOptionParser.cs

```diff
@@ -102,34 +102,26 @@ internal sealed class ExpandOptionParser
         internal int MaxSearchDepth { get; set; }
 
         /// <summary>
-        /// Building off of a PathSegmentToken, continue parsing any expand options (nested $filter, $expand, etc)
-        /// to build up an ExpandTermToken which fully represents the tree that makes up this expand term.
+        /// Building off a PathSegmentToken, continue parsing any select options (nested $filter, $expand, etc)
+        /// to build up an SelectTermToken which fully represents the tree that makes up this select term.
         /// </summary>
-        /// <param name="pathToken">The PathSegmentToken representing the parsed expand path whose options we are now parsing.</param>
-        /// <param name="optionsText">A string of the text between the parenthesis after an expand option.</param>
-        /// <returns>The list of expand term tokens based on the path token, and all available expand options.</returns>
-        internal List<ExpandTermToken> BuildExpandTermToken(PathSegmentToken pathToken, string optionsText)
+        /// <param name="pathToken">The PathSegmentToken representing the parsed select path whose options we are now parsing.</param>
+        /// <param name="optionsText">A string of the text between the parenthesis after a select option.</param>
+        /// <returns>The select term token based on the path token, and all available select options.</returns>
+        internal SelectTermToken BuildSelectTermToken(PathSegmentToken pathToken, string optionsText)

```

Comment:
>BuildSelectTermToken [](start = 33, length = 20)

could more of the implementation of BuildSelectTermToken be shared with BuildExpandTermToken?

Reply:
Maybe. However, i think current codes look more clearly. I'd like to keep them unchanged now.

---
In reply to: [296958543](https://github.com/OData/odata.net/pull/1488#discussion_r296958543) [](ancestors = 296958543)